### PR TITLE
cilium, docs: Add more detail about NFS/SMB dependencies

### PR DESCRIPTION
**Pull request by jrife**
Add more detail about the NFS/SMB problem impacting socket LB, expand on the list of patches that are required to fix the problem, and add a list of kernel versions that contain the patches for Ubuntu and RHEL.

### CODE REVIEW

**Review by aspsk — APPROVED** [review]

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (learnitall) [inline Documentation/network/kubernetes/kubeproxy-free.rst]: I think a period is needed here instead of a comma.
```suggestion
    * NFS and SMB mounts may break when mounted to a ``Service`` cluster IP while using socket-LB.
```

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (learnitall) [inline Documentation/network/kubernetes/kubeproxy-free.rst]: Grammar nit: `system` -> `systems`.

```suggestion
      systems that implement ``ReadWriteMany`` volumes using this pattern. To avoid this problem,
```

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (learnitall) [inline Documentation/network/kubernetes/kubeproxy-free.rst]: Can we add a colon here since we are leading into a list?

```suggestion
      ensure that the following commits are part of your underlying kernel:
```

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (learnitall) [inline Documentation/network/kubernetes/kubeproxy-free.rst]: Same here, could we also use a colon instead?

```suggestion
      These patches have been backported to all stable kernels and some distro-specific kernels:
```

**Review by learnitall — CHANGES_REQUESTED** [review]
Couple of small grammar nits, but otherwise LGTM. Thanks!

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (jrife) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: Done

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (jrife) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: Done

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (jrife) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: Done

File: Documentation/network/kubernetes/kubeproxy-free.rst
Comment (jrife) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: Done

**Comment by jrife** [comment]
@learnitall @aspsk Do we need a release note for this? I'm seeing `needs-release-note-label` on the PR.

**Review by learnitall — APPROVED** [review]
Thanks!

**Comment by learnitall** [comment]
Ah yes, I'll add one :+1:

**Comment by jrife** [comment]
Hi @learnitall the original test run failed and I had to add a word to `Documentation/spelling_wordlist.txt`. Can we retrigger the tests?

**Comment by jrife** [comment]
Bump @learnitall

**Review by learnitall — APPROVED** [review]

**Comment by learnitall** [comment]
Apologies for the delay!

**Comment by jrife** [comment]
@learnitall Looks like some of the workflows are stuck waiting on "approval from a maintainer"?

**Pull request merged**
